File: ScribesF4/Presentation/ngApp/controllers/FriendsController.ts
namespace ScribesF4.Controllers {
    export class FriendsController {
        public friends;
        public users;
        constructor(private $http: ng.IHttpService) {
            this.friends = [{
                name: 'Tom'
            }];
            $http.get('/api/friends')
                .then((response) => {
                    this.friends = response.data;
                });
        }
        public searchUsers(searchTerm) {
            this.users = [];
            if (searchTerm) {
                this.$http.get(`/api/users/${searchTerm}`)
                    .then((response) => {
                        this.users = response.data;
                    });
            }
        }
    }
}

File: ScribesF4/Presentation/Controllers/FriendsController.cs
using ScribesF4.Services;
using ScribesF4.Services.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Http;

namespace ScribesF4.Presentation.Controllers
{
    public class FriendsController : ApiController
    {
        public FriendsService _friendServ;
        public FriendsController(FriendsService friendServ) {
            _friendServ = friendServ;
        }
        //[Authorize]
        //public IList<FriendDTO> GetFriends() {
        //    return _friendServ.ListFriends(User.Identity.Name);
        //}
        [HttpGet]
        [Authorize]
        [Route("api/users/{searchTerm}")]
        public IList<ApplicationUserDTO> SearchUsers(string searchTerm) {
            return _friendServ.ListUsers(searchTerm);
        }
    }
}
